### PR TITLE
[Data][Docs] Fix type hint in "Running Batch Inference with Ray" example

### PR DESCRIPTION
**Pull request by bveeramani**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by ericl — APPROVED** [review]
I've got a bunch of these fixes already in the big one as well, FYI.

File: doc/source/data/doc_code/torch_image_batch_trained.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def __init__(self):  # <1>
         self.model = resnet18(pretrained=True).cuda()
         self.model.eval()
 
-    def __call__(self, batch: List[torch.Tensor]):  # <2>
+    def __call__(self, batch: Dict[str, np.ndarray]):  # <2>
         torch_batch = torch.stack(batch).cuda()  # <3>
```
Comment (amogkam) [inline doc/source/data/doc_code/torch_image_batch_trained.py:41]: don't we need to change this to `torch.stack(batch["images"].tolist())`? stack doesn't work with a dictionary

Comment (ericl) [reply doc/source/data/doc_code/torch_image_batch_trained.py:41]: Yeah I changed this to:

```
    def __call__(self, batch: Dict[str, np.ndarray]) -> Dict[str, np.ndarray]:  # <2>
        torch_batch = torch.stack(batch["preprocessed"]).cuda()  # <3>
```

In here already: https://github.com/ray-project/ray/pull/34880

Waiting to see if tests pass.

**Comment by bveeramani** [comment]
Closed in favor of #34880

**Pull request closed without merging**